### PR TITLE
✨(back) add LimitBrowserCacheTTLHeaders middleware

### PR DESCRIPTION
**Pull request by madmatah**
## Purpose

By default, DjangoCMS sends cached responses with a `Cache-control: max-age` value reflecting the server cache TTL.

The thing is : the server side cache entry get invalidated each time a user publishes new content to the site, but we cannot invalidate our client browser cache entries. That's why we want to set a long TTL on the server side, but a much lower TTL on the browser cache.

## Proposal

- [x] Add LimitBrowserCacheTTLHeaders middleware

### CODE REVIEW

File: src/richie/apps/core/cache.py
Comment (sampaccoud) [inline src/richie/apps/core/cache.py:158]: Maybe cover also the case of invalid values?
```suggestion
try:
    max_ttl = int(max_ttl)
except Exception as err:
    logger.error(err)
    return response
```

And add a test for this case?

File: src/richie/apps/core/cache.py
Comment (sampaccoud) [inline src/richie/apps/core/cache.py:163]:
```suggestion
                # adds it only if it isn't already set
```

File: tests/apps/core/test_cache.py
Comment (sampaccoud) [inline tests/apps/core/test_cache.py:501]: :thinking:

**Review by sampaccoud — APPROVED** [review]

**Comment by madmatah** [comment]
@sampaccoud Thanks for your feedbacks !
I removed 2 unneccessary lines after your last review, i'm ready to merge if it's ok for you :slightly_smiling_face:

File: src/richie/apps/core/cache.py
Comment (sampaccoud) [inline src/richie/apps/core/cache.py:163]: can you really get all these errors int casting?

Comment (madmatah) [reply src/richie/apps/core/cache.py:163]: The `TypeError` and `ValueError` are raised by `int()`
And the `AttributeError` was for the  `getattr` that I forgot to move in the `try:` block :man_facepalming:

Comment (madmatah) [reply src/richie/apps/core/cache.py]: Ok, here is the fixed version, with another test for the case where `MAX_BROWSER_CACHE_TTL` is not defined.

**Review by sampaccoud — APPROVED** [review]

File: src/richie/apps/core/cache.py
Comment (sampaccoud) [inline src/richie/apps/core/cache.py:161]: @madmatah I'm now wondering if it would not have been better to use 600 as default when it is not defined in settings...

Comment (madmatah) [reply src/richie/apps/core/cache.py:161]: IMO this is an opinionated default value. It should be set with full knowledge of the facts.
Maybe, doing nothing silently would have been better ?

Comment (sampaccoud) [reply src/richie/apps/core/cache.py:161]: I think no because you would think that it works. My point of view was that it is no so much opionated as it is a CMS : for everyone, it is the time elapsed between a publication and the moment everyone sees it for sure. 10 minutes seemed a good default for that?

**Pull request merged**
